### DISCUSSION TITLE
Batching for Heterogeneous Graphs

### DISCUSSION THREAD

**Discussion by Angelo92Git** [pyg-team/pytorch_geometric#9127] [General]
Does this library support batching of several heterogeneous graphs, each with the same set of node and edge types?

In particular I am running into problems when I attempt to use the "torch_geometric.data.batch.Batch" class referred to in the tutorial:
https://www.youtube.com/watch?v=Vz5bT8Xw6Dc

How do I create a DataLoader object from heterogeneous graphs?

**Comment by rusty1s** [comment]
Yes, this is fully supported. You can just pass a list of `HeteroData` objects to `DataLoader`.
